fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchURITest.java

```diff
@@ -123,4 +123,92 @@ public void testSearchURI_uri_chained_missing() throws Exception {
         assertSearchReturnsComposition("subject:Basic.missing-uri:missing", "true");
         assertSearchDoesntReturnComposition("subject:Basic.missing-uri:missing", "false");
     }
+
+    @Test
+    public void testSearchURI_canonical() throws Exception {
+        assertSearchReturnsSavedResource("canonical", "http://hl7.org/fhir/DSTU2");
+        assertSearchReturnsSavedResource("canonical", "urn:uuid:53fefa32-1111-2222-3333-55ee120877b7");
+
+        // Matches are supposed to be precise (e.g. case, accent, and escape sensitive), but aren't
+        assertSearchDoesntReturnSavedResource("canonical", "http://hl7.org/fhir/DSTU");
+        assertSearchDoesntReturnSavedResource("canonical", "xttp://hl7.org/fhir/DSTU2");
+        assertSearchDoesntReturnSavedResource("canonical", "urn:uuid:53fefa32-1111-2222-3333-55ee120877b");
+
+        // case tests
+        assertSearchDoesntReturnSavedResource("canonical", "http://HL7.org/FHIR/dstu2");
+        assertSearchDoesntReturnSavedResource("canonical", "urn:uuid:53FEFA32-1111-2222-3333-55EE120877B7");
+
+        // accent tests
+        assertSearchReturnsSavedResource("canonical", "http://åé");
+        assertSearchDoesntReturnSavedResource("canonical", "http://ae");
+        assertSearchDoesntReturnSavedResource("canonical", "http://HL7.org/FHIR/dstü2");
+    }
+
+    @Test
+    public void testSearchURI_canonical_chained() throws Exception {
+        assertSearchReturnsComposition("subject:Basic.canonical", "http://hl7.org/fhir/DSTU2");
+        assertSearchReturnsComposition("subject:Basic.canonical", "urn:uuid:53fefa32-1111-2222-3333-55ee120877b7");
+
+        assertSearchDoesntReturnComposition("subject:Basic.canonical", "http://HL7.org/FHIR/dstu2");
+        assertSearchDoesntReturnComposition("subject:Basic.canonical", "http://hl7.org/fhir/DSTU");
+        assertSearchDoesntReturnComposition("subject:Basic.canonical", "urn:uuid:53FEFA32-1111-2222-3333-55EE120877B7");
+
+        // TODO add test for diacritics and other unusual characters
```

Comment:
Will you do this TODO, or can this comment be removed?

Reply:
removed comment